### DISCUSSION TITLE
元数据自动备份的时候使用内存，不用硬盘做缓存？

### DISCUSSION THREAD

**Discussion by jiyi** [juicedata/juicefs#2983] [General]
使用的时候发现juicefs内存占用突然涨到10G，看了一下日志，发现程序刚刚完成了一个历时11分钟的元数据自动备份。
使用的状态是正在向juicefs文件系统里写文件，15G的缓存文件夹是占满的。正常内存占用小于1G。
所以元数据自动备份的时候数据都存放在内存里面吗？
因为元数据自动备份功能非常好用，未来会做一个单独设置自动备份时使用硬盘做缓存的功能吗？

**Comment by zhijian-pro** [comment]
请问你们的 juicefs 版本 以及 元数据引擎类型？数据量大约多少？

**Reply by jiyi** [reply]
感谢回复
`juicefs version 1.0.2+2022-10-13.514ef03`
使用postgresl 15，参数如下
```
juicefs mount -d \
--buffer-size 1024 \
--cache-dir "/root/cache" \
--cache-size 15240 \
--writeback \
--backup-meta 4h \
-o allow_other \
"postgres://postgres@127.0.0.1:5432/jfs1?sslmode=disable" \
/mnt/jfs
```
数据量的话，今天juicefs_used_inodes 是 5.7401032e+07，那几天有一个手动dump出的文件，大小是5.3G，不知道是否有参考意义。

**Reply by zhijian-pro** [reply]
SQL 引擎确实会这样。如果是 redis 引擎就不会如此。redis 的有优化。

**Reply by zhijian-pro** [reply]
暂时没有自动备份时使用硬盘做缓存的计划，后面可能会进一步优化 SQL 类的内存

**Reply by jiyi** [reply]
我明白了，感谢
